### DISCUSSION TITLE
Log message de-duplication

### DISCUSSION THREAD

**Discussion by rg-gravis** [foxglove/community#1060] [General]
We simultaneously record two bags on our robots: a lightweight automatically-uploaded bag containing only logs (i.e. only the `/rosout` topic, with messages of type [`rcl_interfaces/msg/Log`](https://github.com/ros2/rcl_interfaces/blob/master/rcl_interfaces/msg/Log.msg)), and a much heavier bag with many more topics, but which also includes `/rosout`. We want `/rosout` in both so it can always be seen when doing recording-based playback, but this causes an issue when doing time-range-based playback: all log messages appear twice in the log panel. It would be nice if they were de-duplicated, since the messages are exactly the same.

**Comment by defunctzombie** [comment]
> We want `/rosout` in both so it can always be seen when doing recording-based playback

We support playback of multiple files. So you can select your bag with the `/rosout` topic _and_ your bag with the heavier topics and play them as if they were one bag file. We support this for local files (open file and select both) and for cloud data where you upload both to the same device and all the topics will be accessible as if they were one bag.

**Reply by rg-gravis** [reply]
We do most everything in cloud, and our most common workflow for selecting what to visualize is going to the "devices" tab, selecting a device, finding the recording we want in the list of recordings under that device, and clicking the "visualize" button that appears in the rightmost column of that table. This does not automatically merge bags. 
I'm aware that I can select data using the timeline view to get automatic merging, but I find that to be clunkier when I just want to do a quick visualization of the last run of the bot. Both ways of selecting data have their advantages, and I use both in different circumstances, so de-duplication would be helpful so that no matter which workflow I use for selecting data to visualize, I always only see one copy of each log message.
